src/distributed_ls/ParaSails/DiagScale.c

```diff
@@ -192,7 +192,7 @@ DiagScale *DiagScaleCreate(Matrix *A, Numbering *numb)
         requests, replies_list);
 
     num_replies = FindNumReplies(A->comm, replies_list);
-    free(replies_list);
```

Comment:
@Ram3sh I notice that you changed a bunch of `free` to `hypre_TFree`, which is correct. But there are still a number of places where `free` is used in conjunction with `hypre_TAlloc`. It is not a big problem with standard host malloc/free but with umpire it can be, correct? (allocating with umpire but free with `free`... Is it a problem?).

@rfalgout does it make sense to add regression tests searching for all instances of using `malloc`/`calloc`/`realloc`/`free` and throw errors?

Reply:
I added this when testing host pools. Freeing an Umpire allocation using free is of course very bad. Umpire host pools are very expensive and nobody uses it AFAIK. The Umpire team is working on making it faster, so I have left the code in there for use in the future.

Reply:
I was wondering about these changes myself.  Clearly we didn't change every 'free' in hypre, and I don't remember why (maybe there wasn't actually a reason).  I think it would be fine to change all of it and write a regression test to check that it stays that way.  I would be happy to write it.

Reply:
"very bad" means it would crash, or would screw up umpire's memory pooling algorithms but can still proceed? But in either case, I consider using `free` with `hypre_TAlloc` is a mistake, and we should fix it. Thanks, @Ram3sh

Reply:
Segfault since you will be freeing an allocation somewhere in the middle.

Reply:
> I was wondering about these changes myself. Clearly we didn't change every 'free' in hypre, and I don't remember why (maybe there wasn't actually a reason). I think it would be fine to change all of it and write a regression test to check that it stays that way. I would be happy to write it.

@rfalgout sounds good. Please do, and I will make the code changes. Thanks!

Reply:
Just added a memory usage test that checks for 'malloc', 'calloc', and 'free'.  Let me know if you think I have missed something.  Thanks!